src/coreclr/tools/aot/crossgen2/crossgen2_publish.csproj

```diff
@@ -29,6 +27,7 @@
     <!-- Disable crossgen on FreeBSD when cross building from Linux. -->
     <PublishReadyToRun Condition="'$(TargetOS)' == 'freebsd' and '$(CrossBuild)' == 'true'">false</PublishReadyToRun>
     <PublishReadyToRunComposite>$(PublishReadyToRun)</PublishReadyToRunComposite>
+    <UseLocalAppHostPack>true</UseLocalAppHostPack>
```

Comment:
Can you move this line back up to the unconditional property group? I think this is what's causing the failures (as the local apphost pack can't be found).

Reply:
After reverting, next place where it looks for apphost is:
> .dotnet/sdk/9.0.100-preview.3.24204.13/Microsoft.Common.CurrentVersion.targets(5270,5): error MSB3030: Could not copy the file "/__w/1/s/artifacts/obj/coreclr/crossgen2_publish/linux.x64.Checked/apphost" because it was not found.
(from `Build coreclr Pri0 Runtime Tests Run linux x64 checked failed`)

Reply:
Is crossgen2_publish NativeAOT'd here?

We just updated the repo to Preview 3 SDK that has this fix: https://github.com/dotnet/sdk/pull/38644 - SDK should not even be looking for an apphost when doing NAOT publish.

Reply:
I tried a few things but this looks like a deadlock.
* crossgen2_publish project is manully publishing AOT app using live build, meaning PublishAot=true is not set
* explicitly setting UseAppHost=false with SelfContained=false for NativeAotSupported != '' failed with `error NETSDK1102: Optimizing assemblies for size is not supported for the selected publish configuration. Please ensure that you are publishing a self-contained app`
* explicitly setting SelfContained=false with with SelfContained=true failed with `error NETSDK1067: Self-contained applications are required to use the application host. Either set SelfContained to false or set UseAppHost to true.`
* also PublishTrimmed is requiring apphost

Reply:
Yes, this manual setup is likely there so crossgen2 is published with live build of ILCompiler when NativeAotSupported==true. ILCompiler itself is published using nuget package. crossgen2_publish.csproj line 4 is setting `_IsPublishing`. Invoking `Publish` / `PublishToDisk` directory gives the same missing apphost error. -v:diag dump suggests one of the linker/trimmer target is trying to copy apphost where it fails.